Specification.md

```diff
@@ -0,0 +1,15 @@
+### plugins.yaml
+
+The `plugins.yaml` file contains a list of objects, each representing a plug-in.
```

Comment:
Can you also add a few words here on why yaml was chosen over the ES-internal data-format? (I can think of good reasons and I don't oppose the use of yaml, but let's document the reasons here.)

Reply:
it's the perfect combination of being easily human-readable, and machine-readable. Being human-readable is priority, because compatibility is handled via the `generated` folder, and i didn't feel comfortable with implementing even more ES data parsers.. Did i miss anything?

Reply:
Looks quite complete.
What I was thinking of is that the downside of choosing yaml is that the ES-game-engine itself would require an additional parser or an additional generated/translated file when it also starts reading plugin lists.

For me it is fine if you just mention those two options of making the plugin-lists available to the game-engine and with those remarks we can just cross that bridge when we get there.

Reply:
Idk about you, but in my book it's actually easier to generate the ES datafiles in the CI job and not bother with yaml+cpp at all.

Reply:
Yes, I would also expect generating the ES datafiles in the CI-job to be the easiest solution when we go this route.